sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/agents/SearchKnowledgeAgentClientBuilder.java

```diff
@@ -0,0 +1,264 @@
+     */
+    public SearchKnowledgeAgentClientBuilder() {
+    }
+
+    /**
+     * Sets the service endpoint for the Azure AI Search instance.
+     *
+     * @param endpoint The URL of the Azure AI Search instance.
+     * @return The updated builder object.
+     */
+    @Override
+    public SearchKnowledgeAgentClientBuilder endpoint(String endpoint) {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the agent name for the Azure AI Search agent.
+     *
+     * @param agentName The name of the agent.
+     * @return The updated builder object.
+     */
+    public SearchKnowledgeAgentClientBuilder agentName(String agentName) {
+        this.agentName = agentName;
+        return this;
+    }
+
+    /**
+     * Sets the API version to use for requests.
+     *
+     * @param apiVersion The API version.
+     * @return The updated builder object.
+     */
+    public SearchKnowledgeAgentClientBuilder apiVersion(String apiVersion) {
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder credential(AzureKeyCredential credential) {
+        this.azureKeyCredential = credential;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder credential(TokenCredential credential) {
+        this.tokenCredential = credential;
+        return this;
+    }
+
+    public SearchKnowledgeAgentClientBuilder audience(SearchAudience audience) {
+        this.audience = audience;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder httpLogOptions(HttpLogOptions logOptions) {
+        this.httpLogOptions = logOptions;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder clientOptions(ClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        Objects.requireNonNull(policy, "'policy' cannot be null.");
+        this.perCallPolicies.add(policy); // For simplicity, treat as per-call; refine as needed
+        return this;
+    }
+
+    public SearchKnowledgeAgentClientBuilder serializer(JsonSerializer jsonSerializer) {
+        this.jsonSerializer = jsonSerializer;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder httpClient(HttpClient client) {
+        this.httpClient = client;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder pipeline(HttpPipeline httpPipeline) {
+        this.httpPipeline = httpPipeline;
+        return this;
+    }
+
+    @Override
+    public SearchKnowledgeAgentClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    public SearchKnowledgeAgentClientBuilder retryPolicy(RetryPolicy retryPolicy) {
```

Comment:
Let's break with the other builders in Azure AI Search and not expose `RetryPolicy` directly and force usage of `RetryOptions`.